File: ex11/ex11.cpp
#include <bits/stdc++.h>

using namespace std;

class Solution
{
public:
    int maxArea(vector<int>& height)
    {
        int len = height.size();
        int l = 0, r = len - 1;
        int maxarea = 0;
        while (l < r)
        {
            int high = min(height[l], height[r]);
            int wide = r - l;
            int area = wide * high;
            maxarea = max(area, maxarea);

            if (height[l] < height[r])
            {
                ++l;
            }
            else
            {
                --r;
            }
        }
        return maxarea;

    }
};

int main(void)
{

    return 0;
}

File: ex11/ex11.py
#!/usr/bin/env python3

from typing import List


class Solution:
    def maxArea(self, height: List[int]) -> int:
        l = 0
        r = len(height) - 1
        ans = 0
        while l < r:
            area = min(height[l], height[r]) * (r-l)
            ans = max(area, ans)
            if height[l] < height[r]:
                l += 1
            else:
                r -= 1
        return ans
